File: IN1000/Obliger/Oblig8/old/kolonne først/verden.py
from rutenett import Rutenett

class Verden:
    def __init__(self, rader, kolonner):
        self._rutenett = Rutenett(rader, kolonner)
        self._generasjonsnummer = 0

        self._rutenett.fyll_med_tilfeldige_celler()
        self._rutenett.koble_celler()

    def tegn(self):
        self._rutenett.tegn_rutenett()
        print(f"Generasjon: {self._generasjonsnummer}")
        print(f"Antall levende: {self._rutenett.antall_levende()}")

    def oppdatering(self):

        rutenett_som_liste = self._rutenett._rutenett

        for kolonne in rutenett_som_liste:
            for rad in kolonne:
                rad.tell_levende_naboer()

        for kolonne in rutenett_som_liste:
            for rad in kolonne:
                rad.oppdater_status()

        self._generasjonsnummer += 1

File: IN1010/Oblig/oblig1/Verden.java
class Verden {

    // setter instansvariablene
    private int genNr;
    private Rutenett rutenett;


    // Konstruktør, oppretter rutenett objekt med oppgitt antall rader og kolonner. Setter opp rutenettet for bruk.
    public Verden(int rader, int kolonner) {

        rutenett = new Rutenett(rader, kolonner);
        genNr = 0;

        rutenett.fyllMedTilfeldigeCeller(); 
        rutenett.kobleAlleCeller();
    
    }

    // kjører tegnRutenett() en gang og deretter oppgir informasjon om generasjonsnummer og hvor mange levende som er i rutenettet. 
    public void tegn(){


        rutenett.tegnRutenett();
        System.out.println("Generasjon: " + genNr);
        System.out.println("Antall Levende: " + rutenett.antallLevende());

    }


    // Teller antall levende naboer for hver celle. Deretter oppdaterer den cellens status i følge spillets regler. Oppdaterer generasjons nummeret. 
    
    public void oppdatering(){

        

          for (int rad = 0; rad < rutenett.antRader; rad++) {
            for (int kol = 0; kol < rutenett.antKolonner; kol++) {
        
                rutenett.hentCelle(rad, kol).tellLevendeNaboer();
            }
        }

          for (int rad = 0; rad < rutenett.antRader; rad++) {
            for (int kol = 0; kol < rutenett.antKolonner; kol++) {
        
                rutenett.hentCelle(rad, kol).oppdaterStatus();
            }
        }
        
        genNr++; 
}
}